packages/components/src/components/post-popovercontainer/post-popovercontainer.tsx

```diff
@@ -169,51 +209,103 @@ export class PostPopovercontainer {
+        return {
+          popover: {
+            y: placement === 'top' ? popoverRect.bottom : popoverRect.top,
+            xStart: popoverRect.left,
+            xEnd: popoverRect.right,
+          },
+          trigger: {
+            y: placement === 'top' ? targetRect.top : targetRect.bottom,
+            xStart: targetRect.left,
+            xEnd: targetRect.right,
+          },
+        };
+      } else {
+        // left or right
+        return {
+          popover: {
+            x: placement === 'left' ? popoverRect.right : popoverRect.left,
+            yStart: popoverRect.top,
+            yEnd: popoverRect.bottom,
+          },
+          trigger: {
+            x: placement === 'left' ? targetRect.left : targetRect.right,
+            yStart: targetRect.top,
+            yEnd: targetRect.bottom,
+          },
+        };
+      }
+    };
+
+    const posData = getPositioningData(currentPlacement, popoverRect, targetRect);
+
+    // Clear previous values
+    const propertiesToClear = [
+      '--safe-space-popover-x',
+      '--safe-space-popover-y',
+      '--safe-space-popover-x-start',
+      '--safe-space-popover-x-end',
+      '--safe-space-popover-y-start',
+      '--safe-space-popover-y-end',
+      '--safe-space-trigger-x',
+      '--safe-space-trigger-y',
+      '--safe-space-trigger-x-start',
+      '--safe-space-trigger-x-end',
+      '--safe-space-trigger-y-start',
+      '--safe-space-trigger-y-end',
+    ];
+
+    propertiesToClear.forEach(prop => {
+      this.host.style.removeProperty(prop);
+    });
+
+    if (isVertical) {
+      // For top/bottom placement
+      this.host.style.setProperty('--safe-space-popover-y', `${posData.popover.y}px`);
+      this.host.style.setProperty('--safe-space-popover-x-start', `${posData.popover.xStart}px`);
+      this.host.style.setProperty('--safe-space-popover-x-end', `${posData.popover.xEnd}px`);
+
+      this.host.style.setProperty('--safe-space-trigger-y', `${posData.trigger.y}px`);
+      this.host.style.setProperty('--safe-space-trigger-x-start', `${posData.trigger.xStart}px`);
+      this.host.style.setProperty('--safe-space-trigger-x-end', `${posData.trigger.xEnd}px`);
+    } else {
+      // For left/right placement
+      this.host.style.setProperty('--safe-space-popover-x', `${posData.popover.x}px`);
+      this.host.style.setProperty('--safe-space-popover-y-start', `${posData.popover.yStart}px`);
+      this.host.style.setProperty('--safe-space-popover-y-end', `${posData.popover.yEnd}px`);
+
+      this.host.style.setProperty('--safe-space-trigger-x', `${posData.trigger.x}px`);
+      this.host.style.setProperty('--safe-space-trigger-y-start', `${posData.trigger.yStart}px`);
+      this.host.style.setProperty('--safe-space-trigger-y-end', `${posData.trigger.yEnd}px`);
     }
   }
 
   render() {
     return (
-      <Host data-version={version}>
+      <Host data-version={version} data-safe-space={this.safeSpace}>
```

Comment:
See the comment about reflection on line 68.

```suggestion
      <Host data-version={version}>
```